lib/commands/commands/riot.ts

```diff
@@ -32,39 +34,157 @@ export const getSummonerId = async (ign:string, server:string) => {
+        players[i].lane = participant.timeline.lane;
+    })
+    const ourPlayer = players.find((player:any) => player.id === playerId);
+    const enemies = players.filter((player:any) => player.lane === ourPlayer.lane && player.teamId !== ourPlayer.teamId);
+    const lane = cache["lanes"].find((lane:any) => lane.lane == ourPlayer.lane)
+    const queue = cache["queues"].find((queue:any) => queue.id == lastGameData.data.queueId)
+
+    const embed = new Discord.RichEmbed()
+        .setColor('FDC000')
+        .setThumbnail(lane ? lane.icon : 'https://d1nhio0ox7pgb.cloudfront.net/_img/g_collection_png/standard/256x256/question.png')
+        .setTimestamp(matchBaseData.timestamp)
+        .setFooter(`${queue.map}, ${queue.queue}`, ourPlayer.champion ? ourPlayer.champion.img : lane.icon)
+
+    if (enemies.length === 0) {
+        const content = 'There was no lane opponent in this game.\nEither it was a bot game, one of the laners roamed a lot or one of the laners was AFK.';
+        embed
+            .setTitle(`${ourPlayer.champion ? ourPlayer.champion.name.toUpperCase() : '???'}`)
+            .setDescription(`**${ourPlayer.name}** (${ourPlayer.champion ? ourPlayer.champion.name : '???' })`)
+            .addField('Some info', content)
     }
-    console.log(lastGame.data);    
+    else {
+        const gameFrames:any = {
+            min5: {
+                player: {},
+                enemies: [],
+                allyTeam: 0,
+                enemyTeam: 0,
+            },
+            min10: {
+                player: {},
+                enemies: [],
+                allyTeam: 0,
+                enemyTeam: 0,
+            },
+            min15: {
+                player: {},
+                enemies: [],
+                allyTeam: 0,
+                enemyTeam: 0,
+            }
+        }
+        for (let min = 5; min <= 15; min += 5) {
+            if (lastGameTimeline.data.frames[min]) {
+                const timeline = lastGameTimeline.data.frames[min].participantFrames;
+                const player:any = Object.values(timeline).find((player:any) => player.participantId === ourPlayer.gameId);
+                gameFrames[`min${min}`].player.cs = player.minionsKilled + player.jungleMinionsKilled;
+                gameFrames[`min${min}`].player.gold = player.totalGold;
+                gameFrames[`min${min}`].player.level = player.level;
+
+                enemies.map((enemy:any) => {
+                    const player:any = Object.values(timeline).find((player:any) => player.participantId === enemy.gameId);
+                    gameFrames[`min${min}`].enemies.push({
+                        id: enemy.gameId,
+                        cs: player.minionsKilled + player.jungleMinionsKilled,
+                        gold: player.totalGold,
+                        level: player.level,
+                    })
+                })
+            }
+        }
+
+        const min5 = gameFrames.min5
```

Comment:
Can't this whole block be simplified? I see a lot of repetition and the conditions are really hard to read.

Reply:
Fixed in f81071e